backend/src/main/java/com/bakdata/conquery/io/mina/PipedJacksonProtocolFilter.java

```diff
@@ -0,0 +1,265 @@
+ * - We repeat 3) until {@link IoBuffer#hasRemaining()} returns false, to respect the existence of multiple messages in a single buffer.
+ * 4) If a Reader signals finalization we clear the Sessions reference to it, to force creation of a new one.
+ * 5) The reader receives the {@link IoSession} and {@link org.apache.mina.core.filterchain.IoFilter.NextFilter} as parameters to ensure correct chainig. As a filter can be used by multiple sessions at once.
+ */
+@Slf4j
+@RequiredArgsConstructor
+@ToString(onlyExplicitlyIncluded = true)
+public class PipedJacksonProtocolFilter extends IoFilterAdapter {
+	private static final Thread.Builder THREAD_BUILDER = Thread.ofVirtual().name("%s#reader".formatted(PipedJacksonProtocolFilter.class.getSimpleName()), 0);
+
+	private static final AttributeKey READER_KEY = new AttributeKey(PipedJacksonProtocolFilter.class, "reader");
+	private static final AttributeKey SPILL_BUFFER = new AttributeKey(PipedJacksonProtocolFilter.class, "spill");
+
+	/**
+	 * Max size of a Buffer on my Windows machine was 64Kibi - to enable effective double buffering, we ... double it.
+	 */
+	private static final DataSize PIPED_IO_BUFFER_SIZE = DataSize.kibibytes(2 * 64);
+
+	private final ObjectMapper mapper;
+	private final int initialBufferLength;
+	private final ExecutorService readerThreads = Executors.newCachedThreadPool((runnable) -> {
+		Thread thread = THREAD_BUILDER.unstarted(runnable);
+		thread.setDaemon(true);
+
+		return thread;
+	});
+
+	@Override
+	public void messageReceived(NextFilter nextFilter, IoSession session, Object message) throws Exception {
+		final IoBuffer buffer = (IoBuffer) message;
+
+
+		synchronized (session) {
+			do {
+				Reader asyncReader = getAsyncReader(session);
+
+				if (asyncReader == null) {
+					if (requiresEdgeBuffering(session, buffer)) {
+						// Buffer contains only partial length we need to combine with the next buffer.
+						break;
+					}
+
+					final int remaining = getBufferLength(session, buffer);
+
+					asyncReader = newAsyncReader(remaining, nextFilter, session);
+					log.trace("Starting new reader for {}", DataSize.bytes(remaining));
+				}
+				else {
+					log.trace("Continuing with {}", DataSize.bytes(asyncReader.getRemaining()));
+				}
+
+				final boolean finished = asyncReader.receive(buffer);
+
+				if (finished) {
+					session.setAttribute(READER_KEY, null);
+				}
+
+			} while (buffer.hasRemaining());
+		}
+	}
+
+	private Reader getAsyncReader(IoSession session) {
+		return (Reader) session.getAttribute(READER_KEY);
+	}
+
+	/**
+	 * Test if we have enough contents to read the incoming buffer's length.
+	 * If not, cumulate the remaining buffer into a new buffer.
+	 * <p>
+	 * Incoming buffers will combine their remaining part of the prefixed-length.
+	 */
+	private boolean requiresEdgeBuffering(IoSession session, IoBuffer buffer) throws IOException {
```

Comment:
Die Methode macht mehr als der Name vermuten lässt. Ich würde die fürs verständnis splitten. Z.B. schon in Zeile 118 returnen und dann den Rest im Falle von `true` machen